### PR TITLE
Implemented prime function with assertions

### PR DESCRIPTION
**Pull request by tejasprabhucc**
- In this implementation isPrime function has pre conditions to check for argument is a number and it is greater than 1.
- Using this isPrime function implemented generatePrimeSeries to generate series of prime number .  
- generatePrimeSeries() has two pre-conditions to check parameter count if it is number and greater than 1.
- Post conditions use strictEqual method in assert object to check for returned value.

### CODE REVIEW

**Review by shripada — CHANGES_REQUESTED** [review]
Can you check you PR, I am not seeing anyhting credible. May be you missed to push?

**Comment by tejasprabhucc** [comment]
> Can you check you PR, I am not seeing anyhting credible. May be you missed to push?

Sir I thing the code has been collapsed, could you please expand it.

File: src/prime.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    // primes are always > 1, and they will have a factor other than themselves or 1
+    assert(num >= 1, "num cannot be negative");
+    for (let i = 2; i < num; i++) {
+        if (num % i === 0) {
+            return false;
+        }
+    }
+    return true;
+}
+
```
Comment (shripada) [inline src/prime.js:15]: add assertions for isPrime too

File: src/prime.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    return primeArr;
+}
+
+assert.strictEqual(generatePrimeSeries(-1), [], "Couldn't generate prime series");
+
+assert.strictEqual(generatePrimeSeries(0), [], "Couldn't generate prime series");
+
+assert.strictEqual(generatePrimeSeries(3), [2, 3, 5], "Couldn't generate prime series");
+
+assert.strictEqual(generatePrimeSeries(5), [2, 3, 5, 7, 11], "Couldn't generate prime series");
```
Comment (shripada) [inline src/prime.js:41]: assert message is wrong

Comment (shripada) [reply src/prime.js:41]: Also add asserts, for cases where expect generatePrimes to throw. See `assert.throws`

**Review by shripada — CHANGES_REQUESTED** [review]

**Pull request merged**